prometheus_alerting_consistency.md

```diff
@@ -0,0 +1,103 @@
+# Prometheus Alerting Consistency
+
+## Summary
+
+This guide is originally based on the [Alerting Consistency][1] proposal for OpenShift alerts.
+
+Clear and actionable alerts are a key component of a smooth operational experience.
+Ensuring you have clear and concise guidelines for engineers and SRE creating new alerts will result in a better experience for end users.
+
+## Recommended Reading
+
+A list of references on good alerting practices:
+
+* [Google SRE Book - Monitoring Distributed Systems][2]
+* [Prometheus Alerting Documentation][3]
+* [Alerting for Distributed Systems][4]
+
+## Style Guide
+
+* Alert names MUST be CamelCase, e.g.: `BackendServiceStuck`
+* Alert names SHOULD be prefixed with a component, e.g.: `ZookeeperPersistentVolumeFillingUp`
+  * There may be exceptions for some broadly scoped alerts, e.g.: `TargetDown`
+* Alerts MUST include a `severity` label indicating the alert's urgency.
+  * Valid severities are: `critical` or `warning` — see below for
+    guidelines on writing alerts of each severity.
+* Alerts MUST include `summary` and `description` annotations.
+  * Think of `summary` as the first line of a commit message, or an email
+    subject line.  It should be brief but informative.  The `description` is the
+    longer, more detailed explanation of the alert.
+* Alerts SHOULD include a `namespace` label indicating the source of the alert.
+  * Many alerts will include this by virtue of the fact that their PromQL
+    expressions result in a namespace label.  Others may require a static
+    namespace label.
```

Comment:
Oh, that's interesting. 
I'm initially thinking it's only relevant if doing federation from multiple clusters with a label for the cluster id.
However, it's also useful if you have separate alerts per cluster, but they all eventually page the same SRE team.
The label name could be different for everyone, but we can capture the intention.
How about this?
```suggestion
* Alerts SHOULD include a `namespace` label indicating the source of the alert.
  * Many alerts will include this by virtue of the fact that their PromQL
    expressions result in a namespace label.  Others may require a static
    namespace label.
* Alerts SHOULD include a cluster identifier label, especially in cases where alerts are defined or routed centrally for multiple services & clusters.
  * This can hint to an SRE where the affected service is located. It can also be useful for aggregation & filtering when analysing groups of firing alerts.
```